include/rcpputils/find_and_replace.hpp

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCPPUTILS__FIND_AND_REPLACE_HPP_
+#define RCPPUTILS__FIND_AND_REPLACE_HPP_
+
+#include <memory>
+#include <string>
+
+namespace rcpputils
+{
+
+/// Find and replace all instances of a string with another string.
+/**
+ * \param[in] input The input string.
+ * \param[in] find The substring to replace.
+ * \param[in] replace The string to substitute for each occurrence of `find`.
+ * \return A copy of the input string with all instances of the string `find` replaced with the
+ *   string `replace`.
+ */
+template<
+  class CharT,
```

Comment:
If you default the `CharT` to be `char`, then I think raw string will work? Not sure if this is the "right" thing to do, but I don't see how it would hurt.

Reply:
Doesn't seem to change anything. This [SO post](https://stackoverflow.com/a/20758753) suggests doing something like:

```
template<typename CharT, std::size_t Length>
void find_and_replace( const CharT (&string)[Length] )
{
    find_and_replace(std::string(string));
}
```

I'm not sure if that's more correct.

Reply:
Hmm. I'll play around with it quickly. If I don't see anything we can merge it as-is, but the whole point was to get rid of the `inline`.

Reply:
I pushed a change that I think will work, your solution didn't work because you needed a different `Length` template argument for each literal.